In [2]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
import statistics 
import math

In [3]:
matches = pd.read_csv("/content/sample_data/matches.csv")
matches.head()

id  season       city  ...         umpire1        umpire2 umpire3
0   1    2017  Hyderabad  ...     AY Dandekar       NJ Llong     NaN
1   2    2017       Pune  ...  A Nand Kishore         S Ravi     NaN
2   3    2017     Rajkot  ...     Nitin Menon      CK Nandan     NaN
3   4    2017     Indore  ...    AK Chaudhary  C Shamshuddin     NaN
4   5    2017  Bangalore  ...             NaN            NaN     NaN

[5 rows x 18 columns]

In [4]:
# setting up the table with relevant columns
dropList = ["result","player_of_match","venue","umpire1","umpire2","umpire3"]
matches.drop(labels=dropList, axis=1, inplace=True)
matches.head() 

id  season  ... win_by_runs win_by_wickets
0   1    2017  ...          35              0
1   2    2017  ...           0              7
2   3    2017  ...           0             10
3   4    2017  ...           0              6
4   5    2017  ...          15              0

[5 rows x 12 columns]

In [5]:
matches[pd.isnull(matches['winner'])]

id  season       city  ... winner win_by_runs win_by_wickets
300    301    2011      Delhi  ...    NaN           0              0
545    546    2015  Bangalore  ...    NaN           0              0
570    571    2015  Bangalore  ...    NaN           0              0
744  11340    2019  Bengaluru  ...    NaN           0              0

[4 rows x 12 columns]

In [6]:
matches['winner'].fillna('Draw', inplace=True)
matches[pd.isnull(matches['winner'])]

Empty DataFrame
Columns: [id, season, city, date, team1, team2, toss_winner, toss_decision, dl_applied, winner, win_by_runs, win_by_wickets]
Index: []

In [7]:
matches[pd.isnull(matches['city'])]

id  season city  ...                       winner win_by_runs win_by_wickets
461  462    2014  NaN  ...  Royal Challengers Bangalore           0              7
462  463    2014  NaN  ...             Delhi Daredevils           0              4
466  467    2014  NaN  ...          Chennai Super Kings           7              0
468  469    2014  NaN  ...          Sunrisers Hyderabad           4              0
469  470    2014  NaN  ...          Chennai Super Kings           0              7
474  475    2014  NaN  ...              Kings XI Punjab           0              5
476  477    2014  NaN  ...          Sunrisers Hyderabad          15              0

[7 rows x 12 columns]

In [8]:
matches['city'].fillna('Dubai', inplace=True)
matches[pd.isnull(matches['city'])]

Empty DataFrame
Columns: [id, season, city, date, team1, team2, toss_winner, toss_decision, dl_applied, winner, win_by_runs, win_by_wickets]
Index: []

In [9]:
matches.replace(["Deccan Chargers","Delhi Daredevils"],["Sunrisers Hyderabad","Delhi Capitals"],inplace=True,regex=True)
match1 = matches[((matches.team1=="Delhi Capitals") & (matches.team2=="Mumbai Indians")) | ((matches.team1=="Mumbai Indians") & (matches.team2=="Delhi Capitals"))]
match1.shape[0]

24

In [11]:
mw_mi = 0 
mw_dc = 0
lst= [i for i in match1['winner']] 
print("Win Tracker!")
for i in lst:
  if i=="Delhi Capitals":
    mw_dc += 1
  elif i=='Draw':
    continue
  else:
    mw_mi += 1
  print(str(mw_mi)+" "+str(mw_dc))
print("MI vs DC : "+str(mw_mi)+" "+str(mw_dc))

Win Tracker!
1 0
2 0
3 0
3 1
3 2
3 3
4 3
5 3
6 3
7 3
7 4
7 5
8 5
8 6
8 7
9 7
9 8
10 8
10 9
11 9
11 10
11 11
11 12
12 12
MI vs DC : 12 12


In [12]:
last_3_season = match1[match1.season >= 2017]
last_3_season.groupby('winner').winner.count()

winner
Delhi Capitals    3
Mumbai Indians    3
Name: winner, dtype: int64

Out of 26 matches held between MI and DC compared in two scenarios, MI and DC draws with the both the scenarios.

In [13]:
def statistics_for_lists(lst):
  print("Maximum Value Of List:")
  print(max(lst))
  print("Median of the List:")
  print(statistics.median(lst))
  print("Mean of the List:")
  print(statistics.mean(lst))
  print("75% of the Median is:")
  print(statistics.median_high(lst))
  print("Minimum Value of List:")
  print(min(lst))

In [14]:
deliveries = pd.read_csv("/content/sample_data/deliveries.csv")
deliveries.head()

match_id  inning  ... dismissal_kind fielder
0         1       1  ...            NaN     NaN
1         1       1  ...            NaN     NaN
2         1       1  ...            NaN     NaN
3         1       1  ...            NaN     NaN
4         1       1  ...            NaN     NaN

[5 rows x 21 columns]

In [15]:
dropToBeList = ['inning','is_super_over','bye_runs','legbye_runs','fielder']
deliveries.drop(dropToBeList, axis=1, inplace=True)
deliveries.replace(['Deccan Chargers','Delhi Daredevils'],['Sunrisers Hyderabad','Delhi Capitals'],inplace=True,regex=True)
deliveries['dismissal_kind'].fillna('Not Out',inplace=True)
deliveries.head()

match_id         batting_team  ... player_dismissed  dismissal_kind
0         1  Sunrisers Hyderabad  ...              NaN         Not Out
1         1  Sunrisers Hyderabad  ...              NaN         Not Out
2         1  Sunrisers Hyderabad  ...              NaN         Not Out
3         1  Sunrisers Hyderabad  ...              NaN         Not Out
4         1  Sunrisers Hyderabad  ...              NaN         Not Out

[5 rows x 16 columns]

In [16]:
ballbyball = deliveries[((deliveries.batting_team=="Delhi Capitals") & (deliveries.bowling_team=='Mumbai Indians')) | ((deliveries.batting_team=='Mumbai Indians') & (deliveries.bowling_team=="Delhi Capitals"))]
no_of_matches=list(set([i for i in ballbyball['match_id']]))
no_of_matches.sort()
print(len(no_of_matches))

24


In [17]:
#Q2
kr = ballbyball[(ballbyball.dismissal_kind.isin(['bowled','caught','caught and bowled','lbw'])) & (ballbyball.bowler=='K Rabada')].groupby('match_id').dismissal_kind.count()
kr.describe()

count    4.00000
mean     1.50000
std      0.57735
min      1.00000
25%      1.00000
50%      1.50000
75%      2.00000
max      2.00000
Name: dismissal_kind, dtype: float64

There is on an average, chance of fall of 1 to 2 wickets by Kagiso Rabada!

In [18]:
#Q3
hp = ballbyball[(ballbyball.batsman=='HH Pandya')].groupby('match_id').batsman_runs.sum()
hp.describe()

count     9.000000
mean     13.666667
std      14.291606
min       0.000000
25%       2.000000
50%       5.000000
75%      29.000000
max      32.000000
Name: batsman_runs, dtype: float64

Out of 9 matches Hardik played against DC, his average score is 13 to 29

In [20]:
#Q4
noballs = ballbyball[ballbyball.noball_runs!=0].groupby('match_id').noball_runs.count()
noballs.describe()

count    17.000000
mean      1.588235
std       0.712287
min       1.000000
25%       1.000000
50%       1.000000
75%       2.000000
max       3.000000
Name: noball_runs, dtype: float64

Average no of no-balls lies in between 1 and 2

In [22]:
#Q5
total_scores = ballbyball.groupby('match_id').total_runs.sum()
total_scores.describe()

count     24.000000
mean     314.958333
std       56.916781
min      185.000000
25%      290.750000
50%      326.500000
75%      346.000000
max      404.000000
Name: total_runs, dtype: float64

average scores recorded between mi and dc matches lies between 314 to 346